### ISSUE TITLE
Some repos return 500 after upgrade

### ISSUE THREAD

**Issue by ben-github** [gogits/gogs#3481]
I'm running Debian Jessie and using the packager.io repo to install gogs. I have about 50 repos and everything works fine up to gogs_0.9.71-1470887388.b756806.jessie_amd64.deb but when I update to 
gogs_0.9.71-1471266593.94392a7.jessie_amd64.deb or later, one of my repo's always returns pretty gogs 500 "internal server error" page when I go to the repos main web page. Git push/pulls are also broken with a 500 error. But most of my repos work fine, so I don't know what is special about this one.

Downgrading via dpkg to gogs_0.9.71-1470887388.b756806.jessie_amd64.deb fixes the issue, so it is related to a change right after that one. 

I'm using postgres as the db backend.

**Comment by Unknwon** [comment]
Please read https://github.com/gogits/gogs/blob/master/.github/ISSUE_TEMPLATE.md

**Comment by ben-github** [comment]
Not sure what information I'm missing from the issue template. Git version 2.1.4 I don't know if this issue is reproduceable at try.gogs.io because the problematic repo a private repo that I cannot push to try.gogs.io. Most repos work fine, so I don't know the trigger for it. We don't use ssh+git for submodules, is the issue is #1647.

The error in the gog log is:

2016/08/18 12:00:08 [...routers/repo/view.go:270 Home()] [E] ErrGettingEditorconfig: exit status 128 - fatal: 'refs/heads/master' - not a valid ref

although I think this is the symptom, not the problem. Please let me know if there is more information I can provide.

**Comment by Unknwon** [comment]
Does your repo has master branch?

**Comment by ben-github** [comment]
Ok, I found the problem. The repo in question does not have a master branch, however I have other repos that are also missing a master branch. However, the repo with the problem still had the default branch set to master (which doesn't exist). I downgraded to gogs_0.9.71-1470887388.b756806.jessie_amd64.deb changed the default branch to one that exists and then upgraded to gogs 0.9.71-1471499392.7f7216b.jessie and the repo is working again. 

So while it is a misconfiguration, it is easy to get the default branch to point to a non-existent branch when cloning from another repo so the earlier gogs_0.9.71-1470887388.b756806 behavior (or a warning) seems like better behavior.
